### PR TITLE
feat(Accordion): Accordion alignment: 'large' size; no outline in plain

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
### PR Checklist

- ~[ ] Related to Abstract designs:~
- [x] Related to JIRA ticket: WEB-908
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change

### Description

A few scattered additions / improvements to `<Accordion>`:
* We don't use the `"plain"` theme yet anywhere in the monolith, but its primary consumer in [/about/careers](https://codecademy.com/about/careers) doesn't set `outline`. Removed it here.
    * Added font size adjustments to match what's in the careers page too...
* Added a `size` prop that defaults to `"normal"` but can be set to `"large"` to make the thing increase font and icon size.
    * Large accordions use the heading font with different line-height than body - hence the `padding-top`...
* Marked `onChange` as optional, since it is.

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.7.38-alpha.256386.0<br/>@codecademy/gamut@9.3.1-alpha.256386.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f0c79d282c143574d83b9ae--codecademy-styleguide.netlify.app>https://5f0c79d282c143574d83b9ae--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5f0c79d282c143574d83b9ae>Deploy Logs</a></p>

File: packages/gamut/src/AccordionButton/index.tsx [hunk 1]
```diff
@@ -40,23 +40,33 @@ export const AccordionButton: React.FC<AccordionButtonProps> = ({
...
     >
-      {children}
+      <span className={styles.children}>{children}</span>
       <ChevronDownIcon
         className={cx(
           styles.expansionIcon,
           expanded && styles.expansionIconExpanded
         )}
+        height={iconSize}
+        width={iconSize}
```
Comment (codecaaron) [inline packages/gamut/src/AccordionButton/index.tsx:69]: this can be `size={iconSize}` btw

Comment (JoshuaKGoldberg) [reply packages/gamut/src/AccordionButton/index.tsx:69]: Oh nice, thanks 😄

Comment (JoshuaKGoldberg) [reply packages/gamut/src/AccordionButton/index.tsx:69]: Ooh that needs to come from `gamut-icons`. Doing that.

Comment (codecaaron) [reply packages/gamut/src/AccordionButton/index.tsx:69]: Oh nice didn't even notice.  Also we'll probably have to tweak the size when I merge https://github.com/Codecademy/client-modules/pull/838

Tagging here for reminder

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.7.38-alpha.456017.0<br/>@codecademy/gamut@9.3.1-alpha.456017.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f0c9a525b768e5aa160cae6--codecademy-styleguide.netlify.app>https://5f0c9a525b768e5aa160cae6--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5f0c9a525b768e5aa160cae6>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.7.38-alpha.189a33.0<br/>@codecademy/gamut@9.3.1-alpha.189a33.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f0c9eff65a5b2f01f59cf9d--codecademy-styleguide.netlify.app>https://5f0c9eff65a5b2f01f59cf9d--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5f0c9eff65a5b2f01f59cf9d>Deploy Logs</a></p>

**Comment by JoshuaKGoldberg** [comment]
Grr Codecov - ignoring...

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.7.39-alpha.61d3e6.0<br/>@codecademy/gamut@9.3.2-alpha.61d3e6.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f0cbe578619935acd709d35--codecademy-styleguide.netlify.app>https://5f0cbe578619935acd709d35--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5f0cbe578619935acd709d35>Deploy Logs</a></p>

**Pull request merged**